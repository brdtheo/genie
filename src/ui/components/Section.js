import React from "react";

export default function Section(props) {
  const { children, title, customStyle } = props;

  return (
    <section className={`${customStyle ? customStyle : "pb-8 sm:pb-16"} w-full`}>
      <h2 className="text-white text-xl font-semibold leading-none">{title}</h2>
      {children}
    </section>
  );
}
