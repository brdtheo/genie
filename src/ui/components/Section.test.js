/**
 * @jest-environment jsdom
 */
import * as React from "react";
import renderer from "react-test-renderer";

import Section from "./Section";

describe("Section component", () => {
  it("should render without throwing an error", () => {
    const tree = renderer.create(<Section />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render child element", () => {
    const tree = renderer
      .create(
        <Section>
          <p>test</p>
        </Section>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render when having props passed", () => {
    const tree = renderer
      .create(<Section customStyle="bg-lemon" title="Songs" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
