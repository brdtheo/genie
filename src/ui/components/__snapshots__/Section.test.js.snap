// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component should render child element 1`] = `
<section
  className="pb-8 sm:pb-16 w-full"
>
  <h2
    className="text-white text-xl font-semibold leading-none"
  />
  <p>
    test
  </p>
</section>
`;

exports[`Section component should render when having props passed 1`] = `
<section
  className="bg-lemon w-full"
>
  <h2
    className="text-white text-xl font-semibold leading-none"
  >
    Songs
  </h2>
</section>
`;

exports[`Section component should render without throwing an error 1`] = `
<section
  className="pb-8 sm:pb-16 w-full"
>
  <h2
    className="text-white text-xl font-semibold leading-none"
  />
</section>
`;
